tools/src/main/java/org/apache/kafka/trogdor/workload/PayloadGenerator.java

```diff
@@ -0,0 +1,137 @@
+     * @param keyType type of keys generated
+     */
+    public PayloadGenerator(Integer messageSize, PayloadKeyType keyType) {
+        this(messageSize, keyType, DEFAULT_VALUE_DIVERGENCE_RATIO);
+    }
+
+    /**
+     * @param messageSize key + value size
+     * @param valueDivergenceRatio ratio of how much each next value is different from the previous
+     *                             value. Used to approximately control target compression rate (if
+     *                             compression is used).
+     */
+    public PayloadGenerator(Integer messageSize, PayloadKeyType keyType,
+                            double valueDivergenceRatio) {
+        this.baseSeed = 856;  // some random number, may later let pass seed to constructor
+        this.currentPosition = 0;
+        this.valueDivergenceRatio = valueDivergenceRatio;
+        this.random = new Random(this.baseSeed);
+
+        final int valueSize = (messageSize > keyType.maxSizeInBytes())
+                              ? messageSize - keyType.maxSizeInBytes() : 1;
+        this.recordValue = new byte[valueSize];
+        // initialize value with random bytes
+        for (int i = 0; i < recordValue.length; ++i) {
+            recordValue[i] = (byte) (random.nextInt(26) + 65);
+        }
+        this.recordKeyType = keyType;
+    }
+
+    /**
+     * Returns current position of the payload generator.
+     */
+    public long position() {
+        return currentPosition;
+    }
+
+    /**
+     * Creates record based on the current position, and increments current position.
+     */
+    public ProducerRecord<byte[], byte[]> nextRecord(String topicName) {
+        return nextRecord(topicName, currentPosition++);
+    }
+
+    /**
+     * Creates record based on the given position. Does not change the current position.
+     */
+    public ProducerRecord<byte[], byte[]> nextRecord(String topicName, long position) {
+        byte[] keyBytes = null;
+        if (recordKeyType == PayloadKeyType.KEY_MESSAGE_INDEX) {
+            keyBytes = ByteBuffer.allocate(recordKeyType.maxSizeInBytes()).putLong(position).array();
+        } else if (recordKeyType != PayloadKeyType.KEY_NULL) {
+            throw new UnsupportedOperationException(
+                "PayloadGenerator does not know how to generate key for key type " + recordKeyType);
+        }
+        return new ProducerRecord<>(topicName, keyBytes, nextValue(position));
+    }
+
+    @Override
+    public String toString() {
+        return "PayloadGenerator(recordKeySize=" + recordKeyType.maxSizeInBytes()
+               + ", recordValueSize=" + recordValue.length
+               + ", valueDivergenceRatio=" + valueDivergenceRatio + ")";
+    }
+
+    /**
+     * Returns producer record value
+     */
+    private byte[] nextValue(long position) {
+        // randomize some of the payload to achieve expected compression rate
+        random.setSeed(baseSeed + 31 * position + 1);
```

Comment:
Is there an explanation somewhere on how changing the seed before each `nextValue` call achieves this?